ui/app/src/utils/content.ts

```diff
@@ -124,6 +124,24 @@ export function isImage(path: string): boolean {
   );
 }
 
+export function isItemLockedForMe(item: DetailedItem | SandboxItem | LegacyItem, username: string): boolean {
+  return item ? item.lockOwner && item.lockOwner !== username : true;
+}
+
+export function getComputedEditMode({
+  item,
+  username,
+  editMode
+}: {
+  item: DetailedItem;
+  username: string;
+  editMode: boolean;
+}): boolean {
+  return isItemLockedForMe(item, username) === true || hasEditAction(item.availableActions) === false
```

Comment:
Item could be null
```suggestion
  return isItemLockedForMe(item, username) === true || hasEditAction(item?.availableActions) === false
```

Reply:
it is not in the typo, the component using it should validate the item